doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,806 @@
+
+    Returns:
+        A list of ASICs. Index of an ASIC in the list is the index of the ASIC
+        on the module. Index is 0 based.
+
+        An item in the list is a tuple that includes:
+          - ASIC instance number (indexed globally across all modules of
+            he chassis). This number is used to find settings for the ASIC
+            from /usr/share/sonic/device/platform/hwsku/asic_instance_number/.
+          - ASIC PCI address: It is used by syncd to attach the correct ASIC.
+
+        For example: [('4', '0000:05:00.0'), ('5', '0000:07:00.0')]
+          In this example, from the output, we know the module has 2 ASICs.
+          Item ('4', '0000:05:00.0') describes information about the first ASIC
+          in the module. '4' means it is asic4 in the chassis. Settings for this
+          ASIC is at /usr/share/sonic/device/platform/hwsku/4/.
+          And '0000:05:00.0' is its PCI address.
+```
+
+#### 3.1.5 ModuleBase class new APIs
+
+##### 3.1.5.1 Need for consistent storage and access of DPU reboot cause, state and health
+1.  The smartswitch needs to know the reboot cause for DPUs. Please refer to the CLI section for the various options and their effects when executed on the switch and DPUs. 
+
+    Table shows the frame work for DPU reboot-cause reporting
+
+    | DPU Reboot Cause | HW/SW | End_User_Message_in_DPU_STATE |
+    | --- | --- | --- |
+    | REBOOT_CAUSE_POWER_LOSS | HW | Power failure |
+    | REBOOT_CAUSE_HOST_RESET_DPU | SW | Host lost DPU - Try resetting DPU |
+    | REBOOT_CAUSE_HOST_POWERCYCLE_DPU | SW | Host lost DPU - Power cycled DPU |
+    | REBOOT_CAUSE_SW_THERMAL |	SW | Switch software Powered Down DPU due to DPU temperature failure |
+    | REBOOT_CAUSE_DPU_SELF_REBOOT | SW | DPU Software reboots the DPU |
+
+2. Though the get_oper_status(self) can get the operational status of the DPU Modules, the current implementation only has limited capabilities.
+    * Can only state MODULE_STATUS_FAULT and can't show exactly where in the state progression the DPU failed. This is critical in fault isolation, DPU switch over decision, resiliency and recovery
+    * Though this is platform implementation specific, in a multi vendor use case, there has to be a consistent way of storing and accessing the information.
+    * Store the state progression (Powered, PCIe-Link-Status, Host-DPU Eth-Link-Status, Firmware-Boot_status, OS-Boot-Status, ControlPlane-State, DataPlane-Status) on the host ChassisStateDB.
+    * get_state_info(self) will return an object with the ChassisStateDB data
+    * Potential consumer: Switch CLIs, Utils (install/repair images), HA, LB, Life Cycle Manager 
+    * Use cases: Debuggability, error recovery (reset, power cycle) and fault management, consolidated view of Switch and DPU state
+
+* ChassisStateDB Schema for DPU_STATE
+    ```
+    Table: “DPU_STATE”
+
+    SCHEMA
+    key:  dpu_state:1
+    
+    HMSET dpu_state:1
```

Comment:
These details being on DB is overkill. These can be in some debug log

Reply:
This is the KEY for the DB. Doesn't signify the DPU STATE.